source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRChoiceSetViewDataSource.mm

```diff
@@ -100,24 +87,56 @@ - (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
     return 1;
 }
 
-- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
-{
-    UITableViewCell *cell = nil;
-
+- (void)setImageForCell:(UITableViewCell *)cell choiceSetResourceResolver:(NSObject<ACOIResourceResolver> *)choiceSetResourceResolver indexPath:(NSIndexPath * _Nonnull)indexPath {
     if(_userSelections[[NSNumber numberWithInteger:indexPath.row]] == [NSNumber numberWithBool:YES]){
         if(_isMultiChoicesAllowed) {
-            cell = [tableView dequeueReusableCellWithIdentifier:checkedCheckboxReuseID];
+            if(choiceSetResourceResolver && [choiceSetResourceResolver respondsToSelector:@selector(getCheckBoxButtonSelectedState)])
```

Comment:
Can we simplify the code and nested if-else here? May be return early, may be initialize default images only once and return ?